### ISSUE TITLE
Use channel ranges instead of mstransform and reassembly during casa selfcal

### ISSUE THREAD

**Issue by tjgalvin** [flint-crew/flint#228]
Because of my silly I had overlooked the `spw` option available throughout casa tasks. Specifically, there are mechanisms there that to specify a channel range selection. Additionally there is also an append option available in gaincal to add solutions to an existing solutions table. 

To date I had been using a combination of `mstransform` and `cvel` to rewrite the MS to contain the number of desired spectral windows over which `gaincal` is executed in order to get a frequency-dependent set of solutions (even if they are simply piece wise). Rewriting the MS twice when `nspw>1` is inexpensive for small MSs, but very time consuming for larger ones. 

Change the self-calibration strategy in `gaincal` to instead use channel ranges while appending to a solutions table. Two parts:
- add the loop across channel ranges
- ensure that the solutions table appropriate tracks which channels / frequencies solutions are valid for

This is being tracked in the `selfcal-chan-range` branch. Part one is already implemented here. Provided part two checks out the branch can be merged. If it does not work then likely we would have to extend the corresponding `applysolutions` command and move it to the channel range loop...

**Comment by tjgalvin** [comment]
It seems that an error is being raised when `gaincal` is attempting to append. See:

```
2025-02-06 03:43:05	SEVERE	Calibrater::solve	Caught exception: Error attempting append=T
2025-02-06 03:43:05	SEVERE		Error merging spw=0's meta info
2025-02-06 03:43:05	SEVERE		Exception Reported: Error in Calibrater::solve.
2025-02-06 03:43:05	SEVERE	gaincal::::	Task gaincal raised an exception of class RuntimeError with the following message: Error in Calibrater::solve.
```

The two calls were:

```
casa -c gaincal(vis='/scratch3/gal16b/deep_example/flint_rmconfigdecorator/70462/SB70462.EMU_0000-73.beam01.round1.ms',caltable='/scratch3/gal16b/deep_example/flint_rmconfigdecorator/70462/SB70462.EMU_0000-73.beam01.round1.caltable',spw='*:0~143',solint='60s',gaintype='G',minsnr=0.0,calmode='p',selectdata=True,uvrange='>400m',append=False)


casa -c gaincal(vis='/scratch3/gal16b/deep_example/flint_rmconfigdecorator/70462/SB70462.EMU_0000-73.beam01.round1.ms',caltable='/scratch3/gal16b/deep_example/flint_rmconfigdecorator/70462/SB70462.EMU_0000-73.beam01.round1.caltable',spw='*:144~287',solint='60s',gaintype='G',minsnr=0.0,calmode='p',selectdata=True,uvrange='>400m',append=True)
```

I'll have to consult the casa docs.

**Comment by tjgalvin** [comment]
I have something that I believe to be working, but I note the use of the word working. It involved a call into `gaincal` and `applycal` for each of the SPWs, and appropriately handling of the channel string specification being passed to `gaincal`. 

But it is entirely unclear to me whether this results in a CORRECTED_DATA column that has the appropriate mapping, whether all channels are calibrated (in the case the CORRECTED_DATA is replaced entirely, even with a `spw=` specification etc). Asking around for insights.

**Comment by tjgalvin** [comment]
Speaking with Josh Marvil I believe we are on the same page with the understanding that this should work. Confirmation is being obtained from a dev, but I am confident enough in this to merge it into main. 

So I did in #237
